hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -95,25 +102,87 @@ spec = parallel $ do
 
+  it "does not echo history if client says no" $
+    checkCoverage . monadicIO $ do
+      outputs :: [ServerOutput SimpleTx] <- pick arbitrary
+      monitor $ cover 0.1 (null outputs) "no message when reconnecting"
+      monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
+      monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
+      run . failAfter 5 $ do
+        withFreePort $ \port ->
+          withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
+            -- send arbitrary outputs from the server (which should be ignored by the client)
+            mapM_ sendOutput outputs
+            -- start client that doesn't want to see the history
+            withClient port (defaultPath <> "?history=0") $ \conn -> do
+              -- receive only one message which should be the greeting one
+              receivedGreeting <- replicateM 1 (receiveData conn)
+
+              case traverse Aeson.eitherDecode receivedGreeting of
+                Left{} -> failure $ "Failed to decode messages:\n" <> show receivedGreeting
+                Right timedOutputs -> do
+                  (output <$> timedOutputs) `shouldBe` [greeting]
+
+              -- send another message from server
+              newOutput :: ServerOutput SimpleTx <- generate arbitrary
+              sendOutput newOutput
+
+              -- Receive one more message. The messages we sent
+              -- before client connected are ignored as expected and client can
+              -- see only this last sent message.
+              received <- replicateM 1 (receiveData conn)
+
+              case traverse Aeson.eitherDecode received of
+                Left{} -> failure $ "Failed to decode messages:\n" <> show received
+                Right timedOutputs' -> do
+                  (output <$> timedOutputs') `shouldBe` [newOutput]
+              return ()
+
+  it "outputs tx as cbor if client says so" $
+    monadicIO $ do
+      tx :: SimpleTx <- pick arbitrary
+      run . failAfter 5 $ do
+        withFreePort $ \port ->
+          withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
+            let txValidMessage = TxValid{headId = HeadId "some-head-id", transaction = tx}
+            -- client is able to specify they want tx output to be encoded as CBOR
+            withClient port (defaultPath <> "?tx-output=cbor") $ \conn -> do
+              sendOutput txValidMessage
+              -- receive greetings + one more message
+              received :: [ByteString] <- replicateM 2 (receiveData conn)
+              -- expect the first message to be Greeting
+              (received List.!! 0) ^? key "me" . nonNull
+                `shouldBe` Just (Aeson.Object (fromList [("vkey", Aeson.String "d5bf4a3fcce717b0388bcc2749ebc148ad9969b23f45ee1b605fd58778576ac4")]))
```

Comment:
I would just drop the first message and not check that it is, indeed, greetings as we already have other tests checking that the first message is greetings and it would make it more clear that we want to focus on the cbor formatting here and nothing else.